src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5207,13 +5207,34 @@ protected sealed override ImmutableArray<ISymbol> GetDeclaredSymbolsCore(SyntaxN
 
             CheckSyntaxNode(node);
 
-            if (node.GetInterceptableNameSyntax() is { } nameSyntax && Compilation.TryGetInterceptor(nameSyntax.GetLocation()) is (_, MethodSymbol interceptor))
+            if (node.GetInterceptableNameSyntax() is { } nameSyntax && Compilation.TryGetInterceptor(nameSyntax) is (_, MethodSymbol interceptor))
             {
                 return interceptor.GetPublicSymbol();
             }
 
             return null;
         }
+
+#pragma warning disable RSEXPERIMENTAL002 // Internal usage of experimental API
+        public InterceptableLocation? GetInterceptableLocation(InvocationExpressionSyntax node, CancellationToken cancellationToken)
```

Comment:
It seems the only reason this method is not static is that we check that the syntax node belongs to this semantic model, but we don't otherwise use the semantic model.
This raises a design question: why did we choose to attach the public API to a SemanticModel? Is that we want to allow for using semantic information in the future? Or is it that people generating interceptors are expected to already have a SemanticModel at hand anyways (convenience)?

Reply:
> Is that we want to allow for using semantic information in the future?

Yes. For example, if we wanted to rely on the machinery for producing a "display path" like we do when writing diagnostics out to the command line, we would need information within the compilation to do so.